libpkg/pkg_solve.c

```diff
@@ -1035,18 +1039,34 @@ pkg_solve_sat_problem(struct pkg_solve_problem *problem)
 					}
 				}
 
-				fprintf(sb->fp, "cannot %s package %s, remove it from request? ",
-						var->flags & PKG_VAR_INSTALL ? "install" : "remove", var->uid);
+				// do not ask to remove a vital package - user certainly doesn't want this
+				if (!var->unit->pkg->vital) {
+					fprintf(sb->fp, "cannot %s package %s, remove it from request? ",
+							var->flags & PKG_VAR_INSTALL ? "install" : "remove", var->uid);
 
-				fflush(sb->fp);
-				if (pkg_emit_query_yesno(true, sb->buf)) {
-					var->flags |= PKG_VAR_FAILED;
+					fflush(sb->fp);
+					if (pkg_emit_query_yesno(true, sb->buf)) {
+						var->flags |= PKG_VAR_FAILED;
+					}
+				} else {
+					vitals_failed++;
 				}
 
 				failed++;
 				need_reiterate = true;
 			}
 			xstring_free(sb);
+
+			// if all failed vars are vitals, then we have no way to proceed
+			// give up on installing any non-vital package to let the solver
+			// return an empty plan
+			if (vitals_failed == total_failed) {
+				for (i = 0; i < problem->nvars; i ++) {
```

Comment:
would be nice to emit a notice here so that the user is aware the plan result in removing vital packages which is unacceptable so nothing will be done.